source/core/Events.cpp

```diff
@@ -1048,6 +1048,9 @@ VIREO_FUNCTION_SIGNATUREV(WaitForEventsAndDispatch, WaitForEventsParamBlock)
         Int32 &staticCount = eventInfo->eventStructInfo[eventStructIndex].setCount;
 
         UInt32 msTimeout = timeOutPtr ? *timeOutPtr : -1;
+        if (msTimeout == 0) {
```

Comment:
In line 1058, there is a check: msTimeout > 0 ? ...

Did you try changing this line to msTimeout >= 0 ? ...